clients/java/signalr/src/test/java/HubConnectionTest.java

```diff
@@ -32,7 +35,9 @@ public void RegisteringMultipleHandlersAndBothGetTriggered() throws Exception {
         assertEquals(0.0, value.get(), 0);
 
         hubConnection.start();
-        hubConnection.send("inc");
+
+        mockTransport.receiveMessage("{}" + RECORD_SEPARATOR);
+        mockTransport.receiveMessage("{\"type\":1,\"target\":\"inc\",\"arguments\":[]}" + RECORD_SEPARATOR);
 
         // Confirming that our handler was called and that the counter property was incremented.
         assertEquals(2, value.get(), 0);
```

Comment:
Looks like we're only testing what the client _receives_ here, and not asserting what the client _sends_ to make sure it's correct. The new handshake test does this, but the existing tests haven't been modified to do that. Previously the echo transport did a reasonable job of this because it would only echo back what was actually sent. Now, the client could send absolutely nothing and the test would still pass.

Reply:
I'll get the sent message and verify that as well